dll/win32/atl/atl30.c

```diff
@@ -372,8 +372,8 @@ ATOM WINAPI AtlModuleRegisterWndClassInfoW(_ATL_MODULEW *pm, _ATL_WNDCLASSINFOW
 
         if (!wci->m_wc.lpszClassName)
         {
-            static const WCHAR szFormat[] = {'A','T','L','%','0','8','l','x',0};
-            swprintf(wci->m_szAutoName, szFormat, (UINT_PTR)wci);
+            static const WCHAR szFormat[] = {'A','T','L','%','0','8','x',0};
+            swprintf(wci->m_szAutoName, szFormat, PtrToUint(wci));
```

Comment:
That looks strange that for x64 where pointers would be expressed using 16 digits, they truncate it to make it only 8 digits long...

Reply:
An x64 ptr can't fit in the target string...

Reply:
Because their definition of m_szAutoName (with 14 characters) is wrong (and the one quoted on MSDN as well)
The (almost) correct definition that is x64-compatible is
https://github.com/reactos/reactos/blob/master/sdk/lib/atl/atlwin.h#L1896
(I say "almost" because it's missing a `* 2` to correctly count for the number of digits)
AND, the szFormat string misses a ':' after the "ATL" part (and before the number).

Reply:
I am dealing with the warning only, so that is out of my scope, though I trust your explanation.
In addition, ReactOS has [unmatching definitions](https://git.reactos.org/?p=reactos.git&a=search&h=HEAD&st=grep&s=m_szAutoName[).
I suggest you file an issue and/or submit a patch, to both Wine and ReactOS.

Reply:
Here is the "proof" of what I'm telling:
![image](https://user-images.githubusercontent.com/1969829/72392251-d6812580-372f-11ea-9eec-fa0cb52777bb.png)
(the newer versions of ATL use instead the more portable "%p" formatter that was not completely existing back at the time of ATL 3 on Windows 9x).

Reply:
> I am dealing with the warning only
This is not a good argument at all. Silencing a legitimate warning is worse than leaving it in place, since you're removing the reminder to fix the problem without actually fixing it.

That said, this discussion really does belong at the Wine level; from our perspective this is just a sync, so I see no point in rejecting this.

Reply:
> I suggest you file an issue and/or submit a patch, to both Wine and ReactOS.

Here you go: commits 1a231e1e70472d7f1a3e1d1b498dedaedf864a54 and e410a12242a50e850846058ef5c53652340c1724.
Sending patch upstream should be done by whoever deals with these matters.

Reply:
Then maybe restoring a diff by hand after each sync should be done by whoever deals with creating diffs.

Reply:
> This is not a good argument at all.

But...

> this is just a sync,

That is what my comment meant, somehow.

--

> Sending patch upstream

In this very case, I could do it, after Wine 5.0 freeze ends. If that suits you.

Reply:
I'm handling it, sending also a test to show that the string indeed starts with "ATL:"

Reply:
https://source.winehq.org/patches/data/176693